File: backend/src/main/java/com/example/Flicktionary/domain/series/dto/SeriesCastDto.java
package com.example.Flicktionary.domain.series.dto;

import com.example.Flicktionary.domain.actor.dto.ActorDto;
import com.example.Flicktionary.domain.series.entity.SeriesCast;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.lang.NonNull;

@Getter
@AllArgsConstructor
public class SeriesCastDto {

    @NonNull
    private final ActorDto actor;
    private final String characterName;

    public SeriesCastDto(SeriesCast seriesCast) {
        this.actor = new ActorDto(seriesCast.getActor());
        this.characterName = seriesCast.getCharacterName();
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/series/dto/SeriesCastDto.kt
package com.example.Flicktionary.domain.series.dto

import com.example.Flicktionary.domain.actor.dto.ActorDto
import com.example.Flicktionary.domain.series.entity.SeriesCast

data class SeriesCastDto(
    val actor: ActorDto,
    val characterName: String
) {
    constructor(seriesCast: SeriesCast) : this(
        ActorDto(seriesCast.actor),
        seriesCast.characterName
    )
}
